### DISCUSSION TITLE
i18next server configuration with Netlify's (remix-adapter)

### DISCUSSION THREAD

**Discussion by davegomez** [sergiodxa/remix-i18next#164] [Q&A] [answered]
With **Remix v2** they have changed the way some adapters work. One affected is the **Netlify** adapter, which exports the `handleRequest` as `default` from `app/entry.server.tsx` and collides with the configuration required by **i18next**.

The template also added a `remix.init/entry.server.tsx` that looks the same.

You can see the template [here](https://github.com/netlify/remix-template) with both files [app/entry.server.tsx](https://github.com/netlify/remix-template/blob/main/app/entry.server.tsx) and [remix.init/entry.server.tsx](https://github.com/netlify/remix-template/blob/main/remix.init/entry.server.tsx).

How can I apply the **i18next** entry server required configuration while keeping the **Netlify**'s?

**Comment by sergiodxa** [comment, marked as answer]
I suppose they're exporting that function so they can keep control over the file. You will need to see what their `@netlify/remix-adapter` package exports and replicate that in your code to customize it.

IMO what Netlify is doing here it's not a good idea, the entry.server is part of your app, if they want to provide a customized one they should give you the code when starting an app, and maybe extract to functions what they need you to run, but not hide the whole file.

**Reply by davegomez** [reply]
I agree.

Limiting what you can do with the file doesn't look like a good idea.

They are basically exporting a bunch of functions as default.
